daemon/graphdriver/aufs/aufs.go

```diff
@@ -286,6 +286,9 @@ func (a *Driver) Remove(id string) error {
 	for {
 		mounted, err := a.mounted(mountpoint)
 		if err != nil {
+			if os.IsNotExist(err) {
+				return nil
```

Comment:
The rename after this seems to already handle it. For the other paths after that, I don't know how we could be sure that they do not exist without checking.